AnkiDroid/src/main/java/com/ichi2/anki/ReadText.java

```diff
@@ -337,8 +337,14 @@ public static void buildAvailableLanguages() {
     }
 
 
-    public static void releaseTts() {
-        if (mTts != null) {
+    /**
+     * Request that TextToSpeech is stopped and shutdown after it it no longer being used
+     * by the context that initialized it.
+     * No-op if the current instance of TextToSpeech was initialized by another Context
+     * @param context The context used during {@link #initializeTts(Context, ReadTextListener)}
+     */
+    public static void releaseTts(Context context) {
+        if (mTts != null && mReviewer.get() == context) {
```

Comment:
I think here we should clear the context too to prevent the only version of leak we can reasonably prevent with this ugly-but-quite-practical fix?

```suggestion
        if (mTts != null && mReviewer.get() == context) {
            context = null;
```

...I think?

Reply:
Sorry, I'm not quite sure where the leak is happening here. 
`context` is not being stored anywhere from this function, so it shouldn't be leaking here.
Or if you meant `mReviewer`, that has a type of `WeakReference<Context>`, so it shouldn't retain it here either.

Correct me if I'm wrong about this, I haven't used Java in a few months now 😅

Reply:
I can definitely put up an issue for the 'possible' fix soon. 

Take my solution suggestion with a mountain sized grain of salt though, there's a whole lot more about the architecture here and TextToSpeech I'd need to learn about before I'd be confident about it.

Reply:
If `stop` or `shutdown` takes time to return, then `context` is kept in scope and so maybe can not be garbage collected. I do not know about java internal enough to know whether the garbage collector know that `contex` won't be used anymore, and I really really wish that we don't have to deal with this level of details.
Is there any reason to expect that one of those methods take time to return Mike? In particular that it takes so much time that it's a real concern?